diff --git a/libmp3lame/presets.c b/libmp3lame/presets.c
--- a/libmp3lame/presets.c
+++ b/libmp3lame/presets.c
@@ -34,9 +34,10 @@
+/* Studio Duo, 2026-10-10: preset getters may return integer values. */
 #define SET_OPTION(opt, val, def) if (enforce) \
     (void) lame_set_##opt(gfp, val); \
-    else if (!(fabs(lame_get_##opt(gfp) - def) > 0)) \
+    else if (!(fabs((double) (lame_get_##opt(gfp) - (def))) > 0)) \
     (void) lame_set_##opt(gfp, val);
 
 #define SET__OPTION(opt, val, def) if (enforce) \
     lame_set_##opt(gfp, val); \
-    else if (!(fabs(lame_get_##opt(gfp) - def) > 0)) \
+    else if (!(fabs((double) (lame_get_##opt(gfp) - (def))) > 0)) \
     lame_set_##opt(gfp, val);
